### PR TITLE
Add message to rollback and restructre rollback commands

### PR DESCRIPTION
**Pull request by jspspike**
Author has included the following, where applicable:

- [x] Tests
- [x] Changeset

Reviewer has performed the following, where applicable:

- [ ] Checked for inclusion of relevant tests
- [ ] Checked for inclusion of a relevant changeset
- [ ] Checked for creation of associated docs updates
- [ ] Manually pulled down the changes and spot-tested

Fixes # [insert issue number].

### CODE REVIEW

File: packages/wrangler/src/__tests__/deployments.test.ts [hunk 1]
```diff
@@ -247,7 +277,12 @@ describe("deployments", () => {
 					result: true,
 				});
 
-				await runWrangler("deployments rollback 3mEgaU1T-Intrepid-someThing");
+				mockPrompt({
+					text: "Please provide a message for this rollback (280 characters max)",
```
Comment (lrapoport-cf) [inline packages/wrangler/src/__tests__/deployments.test.ts:281]: i think we agreed on 120 char limit
```suggestion
					text: "Please provide a message for this rollback (120 characters max)",
```

File: packages/wrangler/src/deployments.ts [hunk 2]
```diff
@@ -179,6 +185,18 @@ export async function rollbackDeployment(
 		return;
 	}
 
+	const rollbackReason = await prompt(
+		"Please provide a message for this rollback (280 characters max)",
```
Comment (lrapoport-cf) [inline packages/wrangler/src/deployments.ts:189]:
```suggestion
		"Please provide a message for this rollback (120 characters max)",
```

**Comment by lrapoport-cf** [comment]
hey @jspspike , a couple things to address from a functionality perspective: 

- [ ] supplying the `--message` to `wrangler rollback` should cause a bypass of the interactive prompts. currently this doesn't seem to happen.
- [ ] where do we actually enforce character limit for the `message`? i am able to provide a message longer than 280 chars, which is currently specified as the limit. i made notes where we need to change to say the limit is 120, but we also need to update the actual functionality to limit to 120
- [ ] `wrangler deployments view` should show metadata for most recent deployment, but it looks like there's an indexing issue because it's showing data for not the most recent (see example image below)
image

- [ ] similarly to the point above, `wrangler rollback` should cause a roll back to the last deployment, but the rollback is instead happening for some other deployment (see example image below) 
image

- [ ] the output from `wrangler deployments list` should include the contents of the `message` annotation (i know this may currently be missing because changes aren't merged in the backend yet to support, but i'm noting this here to make sure to double check)

File: packages/wrangler/src/__tests__/deployments.test.ts [hunk 3]
```diff
@@ -277,12 +312,14 @@ describe("deployments", () => {
...
 				setIsTTY(false);
 
-				await runWrangler("deployments rollback 3mEgaU1T-Intrepid-someThing");
+				await runWrangler("rollback 3mEgaU1T-Intrepid-someThing");
 				expect(std.out).toMatchInlineSnapshot(`
 			"🚧\`wrangler deployments\` is a beta command. Please report any issues to https://github.com/cloudflare/workers-sdk/issues/new/choose
 
 			? This deployment 3mEgaU1T will immediately replace the current deployment and become the active deployment across all your deployed routes and domains. However, your local development environment will not be affected by this rollback. Note: Rolling back to a previous deployment will not rollback any of the bound resources (Durable Object, R2, KV, etc.).
 			🤖 Using default value in non-interactive context: yes
+			? Please provide a message for this rollback (280 characters max)
```
[the remark this answers is not in the document]
Comment (JacobMGEvans) [reply packages/wrangler/src/__tests__/deployments.test.ts:321]: This part is automated, after the other changes are made.

**Comment by jspspike** [comment]
@lrapoport-cf 
>supplying the `--message` to` wrangler rollback` should cause a bypass of the interactive prompts. currently this doesn't seem to happen.

This is a bug, will fix this

>where do we actually enforce character limit for the `message`? i am able to provide a message longer than 280 chars, which is currently specified as the limit. i made notes where we need to change to say the limit is 120, but we also need to update the actual functionality to limit to 120

>he output from wrangler deployments list should include the contents of the message annotation (i know this may currently be missing because changes aren't merged in the backend yet to support, but i'm noting this here to make sure to double check)

Both of these are done in the backend, I will change the limit to 120 and check once the backend changes are in

>`wrangler deployments view` should show metadata for most recent deployment, but it looks like there's an indexing issue because it's showing data for not the most recent (see example image below)

Looking into this there's a bug in the backend logic for choosing the latest, I will put up a PR for this

>similarly to the point above, `wrangler rollback` should cause a roll back to the last deployment, but the rollback is instead happening for some other deployment (see example image below)

This is working as intended, if you're rolling back you wouldn't want to rollback to the latest deployment (this would do nothing since it's already the active deployment). You would want to rollback to the previous deployment, or the second latest deployment

**Comment by JacobMGEvans** [comment]
> This is working as intended, if you're rolling back you wouldn't want to rollback to the latest deployment (this would do nothing since it's already the active deployment). You would want to rollback to the previous deployment, or the second latest deployment

This I agree with, the `view` is definitely "latest" but the default for `rollback` would be "previous" to the "latest"

**Comment by lrapoport-cf** [comment]
> > This is working as intended, if you're rolling back you wouldn't want to rollback to the latest deployment (this would do nothing since it's already the active deployment). You would want to rollback to the previous deployment, or the second latest deployment
> 
> This I agree with, the `view` is definitely "latest" but the default for `rollback` would be "previous" to the "latest"

@jspspike @JacobMGEvans oh yes you're totally right, my bad!

**Comment by lrapoport-cf** [comment]
> > supplying the `--message` to` wrangler rollback` should cause a bypass of the interactive prompts. currently this doesn't seem to happen.
> 
> This is a bug, will fix this
> 
> >

@jspspike thanks for fixing this! one more small adjustment: if you do `wrangler rollback --message " "` you can bypass the prompt but empty string i.e. `wrangler rollback --message ""` still asks for a prompt. can you change so that this also causes the bypass?

**Review by JacobMGEvans — APPROVED** [review]
Tentatively approved, based on @lrapoport-cf comments and functionality is resolved.

**Comment by lrapoport-cf** [comment]
@jspspike this is looking great, thank you for addressing all the items! one last thing i noticed: on `wrangler deployments view`,  the message field content isn't properly left-aligned with the other fields (see image below). can we fix the alignment? other than that, lgtm!

image

**Review by lrapoport-cf — APPROVED** [review]

**Pull request merged**